de.metas.report/report-service/src/main/java/de/metas/adempiere/report/jasper/JasperEngine.java

```diff
@@ -132,41 +145,60 @@ private JasperPrint createJasperPrint(final ReportContext reportContext) throws
 		final Map<String, Object> jrParameters = createJRParameters(reportContext);
 		final JasperReport jasperReport = createJasperReport(ctx, reportContext.getAD_Process_ID(), jrParameters, jasperLoader);
 
-		Connection conn = null;
-		try
+		// JSON Data source
+		if (isJasperJSONReport(reportContext))
 		{
 			//
-			// Create jasper's JDBC connection
-			conn = getConnection();
-			final String sqlQueryInfo = "jasper main report=" + jasperReport.getProperty(JRPROPERTY_ReportPath)
-					+ ", AD_PInstance_ID=" + reportContext.getPinstanceId();
+			// Fill the report
+			final JasperPrint jasperPrint = ADJasperFiller.getInstance().fillReport(jasperReport, jrParameters, jasperLoader);
+			return jasperPrint;
 
-			final String securityWhereClause;
-			if (reportContext.isApplySecuritySettings())
+		}
+		else
+		{
+			Connection conn = null;
+			try
 			{
-				final IUserRolePermissions userRolePermissions = reportContext.getUserRolePermissions();
-				final String tableName = reportContext.getTableNameOrNull();
-				securityWhereClause = userRolePermissions.getOrgWhere(tableName, Access.READ);
+
+				//
+				// Create jasper's JDBC connection
+				conn = getConnection();
+				final String sqlQueryInfo = "jasper main report=" + jasperReport.getProperty(JRPROPERTY_ReportPath)
+				+ ", AD_PInstance_ID=" + reportContext.getPinstanceId();
+
+				final String securityWhereClause;
+				if (reportContext.isApplySecuritySettings())
+				{
+					final IUserRolePermissions userRolePermissions = reportContext.getUserRolePermissions();
+					final String tableName = reportContext.getTableNameOrNull();
+					securityWhereClause = userRolePermissions.getOrgWhere(tableName, Access.READ);
+				}
+				else
+				{
+					securityWhereClause = null;
+				}
+
+				final JasperJdbcConnection jasperConn = new JasperJdbcConnection(conn, sqlQueryInfo, securityWhereClause);
+
+				//
+				// Fill the report
+				final JasperPrint jasperPrint = ADJasperFiller.getInstance().fillReport(jasperReport, jrParameters, jasperConn, jasperLoader);
+				return jasperPrint;
 			}
-			else
+			finally
 			{
-				securityWhereClause = null;
+				DB.close(conn);
+				conn = null;
 			}
-
-			final JasperJdbcConnection jasperConn = new JasperJdbcConnection(conn, sqlQueryInfo, securityWhereClause);
-
-			//
-			// Fill the report
-			final JasperPrint jasperPrint = ADJasperFiller.getInstance().fillReport(jasperReport, jrParameters, jasperConn, jasperLoader);
-			return jasperPrint;
-		}
-		finally
-		{
-			DB.close(conn);
-			conn = null;
 		}
 	}
 
+	private boolean isJasperJSONReport(final ReportContext reportContext)
+	{
+		return X_AD_Process.TYPE_JasperReportsJSON.equals(reportContext.getType())
+				&& !Check.isEmpty(reportContext.getJSONPath(), true);
```

Comment:
if would assume the context is already valid.
i.e. if the Type is JSON, at this point we shall have the JSON Path set for sure.